Documentation/building/unix-instructions.md

```diff
@@ -1,16 +1,90 @@
 Building CoreFX on FreeBSD, Linux and OS X
 ==========================================
 
-Building CoreFX is pretty straightforward. Clone the repo and run the build script.  For specific steps to build on Ubuntu, [navigate to the end of this document](#steps-to-build-on-ubuntu).
+The CoreFX build has two logical components, the native build which produces the
+"shims" (which provide a stable interface between the OS and managed code) and
+the managed build which produces the MSIL code and nuget packages that make up
+CoreFX.
+
+The native component should be buildable on any system, but the managed
+components require a version of the .NET Core CLI (which the build will
+download) so managed components can only be built on a subset of distros.
+
+### Prerequsites (native build)
+
+The native build produces shims over libc, openssl, libcurl and libz. The build
+system uses CMake (2.8.12 or higher) to generate Makefiles using clang (3.5 or
+higher).  The build also uses git for generating some version information.
```

Comment:
The version information stuff depends on it.  It's not really "light up".  It's possible that the build succeeds without it but the version information would be incorrect and its also possible that the build just fails.  I haven't given it a shot.